mbed_build/_internal/config/cumulative_data.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Ability to parse cumulative attributes from Sources."""
+import itertools
+import re
+from dataclasses import dataclass, field, fields
+from typing import Any, Iterable, Set, Tuple
+
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mbed_build._internal.config.source import Source
+
+
+@dataclass
+class CumulativeData:
```

Comment:
Mm cumulative is a nicer term than accumulating. I'm tempted to update in mbed-targets. I'll see if I have time.